src/openrpc/lifecycle.json

```diff
@@ -1,379 +1,626 @@
+                {
+                    "name": "Default Example",
+                    "params": [
+                        {
+                            "name": "listen",
+                            "value": true
+                        }
+                    ],
+                    "result": {
+                        "name": "Default Result",
+                        "value": {
+                            "state": "suspended",
+                            "previous": "hibernated"
+                        }
+                    }
+                }
+            ]
         },
         {
-          "name": "capabilities",
-          "x-uses": ["xrn:firebolt:capability:lifecycle:state"]
-        }
-      ],
-      "summary": "Listen to the unloading event",
-      "params": [
-        {
-          "name": "value",
-          "schema": {
-            "$ref": "#/components/schemas/LifecycleEvent"
-          }
+            "name": "destroy",
+            "tags": [
+                {
+                    "name": "notifier",
+                    "x-notifier": "Lifecycle.destroy",
+                    "x-event": "Lifecycle.onDestroy"
+                },
+                {
+                    "name": "capabilities",
+                    "x-uses": [
+                        "xrn:firebolt:capability:lifecycle:state"
+                    ]
+                }
+            ],
+            "summary": "Listen to the destroy event \n\nHowever, this event is unreliable and app may not get this notification in the event of platform needs to kill the app quickly",
+            "params": [
+                {
+                    "name": "listen",
+                    "required": true,
+                    "schema": {
+                        "type": "boolean"
+                    }
+                }
+            ],
+            "result": {
+                "name": "value",
+                "schema": {
+                    "$ref": "#/components/schemas/DestroyEvent"
+                }
+            },
+            "examples": [
+                {
+                    "name": "Default Example",
+                    "params": [
+                        {
+                            "name": "listen",
+                            "value": true
+                        }
+                    ],
+                    "result": {
+                        "name": "Default Result",
+                        "value": {
+                            "reason": "userExit",
+                            "previous": "paused"
+                        }
+                    }
+                }
+            ]
         }
-      ],
-      "examples": [
-        {
-          "name": "Default Example",
-          "params": [
-            {
-              "name": "value",
-              "value": {
-                "state": "unloading",
-                "previous": "inactive"
-              }
+    ],
+    "components": {
+        "schemas": {
+            "LifecycleEvent": {
```

Comment:
This schema can go away. None of the Lifecycle events have parameters now, except for onDestroy.
We don't tell the app the "previous" and current state, because the event names are unique depending on where you came from.